B-17 P0 · RAILWAY_TOKEN_ACC1/ACC2 expired — ALL healer workflows blocked at Sanity-check tokens

# B-17 · RAILWAY_TOKEN_ACC1 / ACC2 expired — ALL healer workflows blocked

## R5-verified evidence (2026-05-14 06:01 UTC, cycle 14)

### Symptom

Writer service `ssot.bpb_samples` frozen for **10h 34m**:

| Probe | Value |
|---|---|
| `MAX(ts)` on SHORT-WAVE-MATRIX canons | `2026-05-13T19:26:40.738Z` |
| `stale_sec` | 38095 |
| `rows_1h` | 0 |
| `rows_6h` | 0 |
| `rows_24h` | 877 (all before 19:26 UTC) |
| Non-matrix canons last_ts | `2026-05-12T17:00:38.059Z` (stale 37h) |

### Root cause

Every healer workflow that calls Railway team GraphQL fails at the **first** step (`Sanity-check tokens`):

```
##[error]token ACC1 failed sanity check
##[error]Process completed with exit code 3.
```

The sanity check curls `https://backboard.railway.com/graphql/v2` with `RAILWAY_TOKEN_ACC1` and expects `.data.projects.edges` — receives empty/null instead, meaning the team token is **revoked or expired** on Railway side.

### Affected workflow runs (R5-verified)

| Workflow | Latest run | Conclusion | Failed step |
|---|---|---|---|
| `writer-env-fix.yml` | [25844596039](https://github.com/gHashTag/trios-railway/actions/runs/25844596039) (2026-05-14 06:00 UTC, dispatch this cycle) | ❌ failure | Sanity-check tokens |
| `writer-env-fix.yml` | [25843795652](https://github.com/gHashTag/trios-railway/actions/runs/25843795652) (2026-05-14 05:35 UTC) | ❌ failure | Sanity-check tokens |
| `mcp-emergency-redeploy.yml` | [25843877778](https://github.com/gHashTag/trios-railway/actions/runs/25843877778) (2026-05-14 05:38 UTC) | ❌ failure | — |

### Last good token use

- `refresh-acc47.yml` run [25754312720](https://github.com/gHashTag/trios-railway/actions/runs/25754312720) at **2026-05-12 18:31 UTC** = ✅ success
- Token last rotated in repo secrets: **2026-05-12 14:44:43 UTC** (`RAILWAY_TOKEN_ACC1`)

→ Token worked for ~4 hours after rotation, then stopped. Likely:
  1. Team membership change on Railway side revoked the token, OR
  2. Token expired (Railway team tokens have ≤ 30-day TTL by default), OR
  3. Token scope was narrowed and `projects { edges }` query no longer returns rows.

### Why this is a hard blocker

- Layer-2 Doctor (`doctor-loop.yml`) — PR #155 still draft, doctor not deployed yet
- Gardener-loop v2.12 (post-#149 merge) reports `success` on every tick but **wakes 0 effective trainers** — gas-lighting persists (B-12 / B-16 pattern)
- Stage 2d in gardener-loop run [25843994908](https://github.com/gHashTag/trios-railway/actions/runs/25843994908) claimed `⚡ FORCED WAKE-UP: 159 redeployed · 1 failed` — but writer still dead. Without working Railway tokens, even `serviceInstanceDeployV2` calls have no effect (Railway accepts the GraphQL mutation, then deployments queue but never apply because no team auth).

### Cure (operator-only)

```bash
# On Railway dashboard, rotate team tokens for all 7 ACC1-7 accounts,
# OR generate new project-access-tokens scoped to IGLA + ACC2-RP projects.
# Then update repo secrets:

gh secret set RAILWAY_TOKEN_ACC1 --repo gHashTag/trios-railway --body "$NEW_TOKEN_ACC1"
gh secret set RAILWAY_TOKEN_ACC2 --repo gHashTag/trios-railway --body "$NEW_TOKEN_ACC2"
# ... ACC3..ACC7 similarly

# Verify with sanity-check workflow dispatch:
gh workflow run writer-env-fix.yml --repo gHashTag/trios-railway -f confirm=PHI
gh run watch  # expect Sanity-check tokens ✅
```

### Verification gates (post-fix)

1. `writer-env-fix.yml` workflow_dispatch → ✅ success (currently fails at Sanity-check)
2. `ssot.bpb_samples` writes resume → `stale_sec < 600` within T+15min
3. Gardener-loop Stage 2d `wake_effective=true` on next hourly tick

### Anchors

- Anchor: `φ² + φ⁻² = 3 · TRINITY · NEVER STOP`
- DOI: 10.5281/zenodo.19227877
- Cycle 14 NASA report: `DOCTOR-CYCLE-14-RVR-001`

cc: B-12 (#151 gas-lighting), B-16 (Stage 2d claim vs reality), PR #149 (merged), PR #155 (doctor Layer-2 pending)

## R5 verification @ 06:15 UTC — ACC1/ACC2 ротированы, но всё ещё failure

### Что проверено
- `RAILWAY_TOKEN_ACC1` updated_at = **2026-05-14T06:05:28Z** ✅ ротирован
- `RAILWAY_TOKEN_ACC2` updated_at = **2026-05-14T06:04:54Z** ✅ ротирован
- ACC3..ACC7 — **НЕ ротированы** (всё ещё 2026-05-12 14:45-14:46 UTC)

### Re-dispatch результат
- [Run 25845100317](https://github.com/gHashTag/trios-railway/actions/runs/25845100317) @ 2026-05-14 06:15 UTC = ❌ failure
- Failed step: `Sanity-check tokens` → `##[error]token ACC1 failed sanity check / exit code 3`
- Symptom: новый ACC1 token тоже возвращает пустой `.data.projects.edges` от `https://backboard.railway.com/graphql/v2`

### URL endpoint sanity (R5)
Оба canonical URL отвечают HTTP 200:
- `backboard.railway.com/graphql/v2` ✅
- `backboard.railway.app/graphql/v2` ✅

→ Проблема **не в URL**, а в **типе токена**.

### Hypothesis (R5-honest)
Workflow `writer-env-fix.yml` (видно в [.github/workflows/writer-env-fix.yml#L52-L60](https://github.com/gHashTag/trios-railway/blob/main/.github/workflows/writer-env-fix.yml)) ставит:
```yaml
env:
  TOKEN_ACC1: ${{ secrets.RAILWAY_TOKEN_ACC1 }}
  RAILWAY_TOKEN_AUTH: team
```

И шлёт:
```bash
curl ... -H "Authorization: ${T}" ...
-d '{"query":"{ projects { edges { node { id name } } } }"}'
```

Запрос `{ projects { edges } }` — это **team/account-level scope**. Если оператор сгенерил **project-access-token** (project-scoped, который привязан к одному проекту), Railway вернёт `data.projects.edges = null` потому что project-token не имеет доступа к списку проектов.

### Что нужно (R5 fix recipe)

Зайти в Railway dashboard → **Account Settings → Tokens** (НЕ Project Settings → Tokens):
1. Создать **Account Token** (либо **Team Token** если есть team plan), scope = `team` или `account`. Этот токен может запросить `{ projects { edges } }`.
2. Обновить repo secret: `gh secret set RAILWAY_TOKEN_ACC1 --repo gHashTag/trios-railway --body "<account_token_acc1>"`
3. То же для `RAILWAY_TOKEN_ACC2`.
4. (Опционально) обновить ACC3..ACC7 если они нужны для других workflow.
5. Re-dispatch: `gh workflow run writer-env-fix.yml --repo gHashTag/trios-railway -f confirm=PHI`
6. Ожидание: step `Sanity-check tokens` → `token ACC1 OK / token ACC2 OK`.

### Writer pulse (не сдвинулся)
| Probe | Value |
|---|---|
| `MAX(ts)` SHORT-WAVE-MATRIX | `2026-05-13T19:26:40.738Z` |
| `stale_sec` | 38906 (10h 48m) |
| `rows_15m` | 0 |
| `rows_1h` | 0 |

Anchor: `φ² + φ⁻² = 3 · TRINITY · NEVER STOP`

--- a/.github/workflows/writer-env-fix.yml
+++ b/.github/workflows/writer-env-fix.yml
@@ -56,7 +56,7 @@ jobs:
     env:
       TOKEN_ACC1: ${{ secrets.RAILWAY_TOKEN_ACC1 }}
       TOKEN_ACC2: ${{ secrets.RAILWAY_TOKEN_ACC2 }}
-      RAILWAY_TOKEN_AUTH: team
+      RAILWAY_TOKEN_AUTH: project  # B-18 fix (2026-05-14): all ACC1..7 tokens are project-scoped (PAT), not team. Verified by token-classify.yml run 25846091578.
       IGLA_PROJECT_ID: e4fe33bb-3b09-4842-9782-7d2dea1abc9b
       ACC2_PROJECT_ID: 12c508c7-1196-468d-b06d-d8de8cb77e93
       MCP_SERVICE_ID: db786a4b-5a79-4643-b915-e9184680cf97
@@ -78,18 +78,21 @@ jobs:
             echo "::error::RAILWAY_TOKEN_ACC2 secret is empty"
             exit 2
           fi
+          # B-18 fix: use Project-Access-Token (PAT) header + project-scoped query.
+          # The old Authorization: Bearer + { projects { edges } } pattern
+          # required an account/team token; all ACC1..7 secrets are PAT.
           for L in ACC1 ACC2; do
             T_VAR="TOKEN_${L}"
             T="${!T_VAR}"
             curl -sS -X POST https://backboard.railway.com/graphql/v2 \
                 -H "Content-Type: application/json" \
-                -H "Authorization: Bearer $T" \
-                -d '{"query":"{ projects { edges { node { id name } } } }"}' \
-                | jq -e '.data.projects.edges' > /dev/null || {
-                  echo "::error::token $L failed sanity check"
+                -H "Project-Access-Token: $T" \
+                -d '{"query":"{ projectToken { project { id name } } }"}' \
+                | jq -e '.data.projectToken.project.id' > /dev/null || {
+                  echo "::error::token $L failed sanity check (Project-Access-Token mode)"
                   exit 3
                 }
-            echo "token $L OK"
+            echo "token $L OK (PAT scope)"
           done
 
       - name: Apply fix to all reachable services
@@ -137,9 +140,12 @@ jobs:
           resolve_env() {
             local TOKEN="$1"
             local PROJECT_ID="$2"
+            # B-18 fix: PAT header instead of Authorization: Bearer.
+            # NOTE: project(id) query is implicit for PAT — the token already
+            # carries the project, but Railway still accepts an explicit projectId.
             curl -sS -X POST https://backboard.railway.com/graphql/v2 \
                 -H "Content-Type: application/json" \
-                -H "Authorization: Bearer $TOKEN" \
+                -H "Project-Access-Token: $TOKEN" \
                 -d "{\"query\":\"query env(\$projectId: String!){ project(id:\$projectId){ environments { edges { node { id name } } } } }\",\"variables\":{\"projectId\":\"$PROJECT_ID\"}}" \
                 | jq -r '.data.project.environments.edges[] | select(.node.name=="production") | .node.id' \
                 | head -n 1
@@ -150,9 +156,10 @@ jobs:
           fetch_services() {
             local TOKEN="$1"
             local PROJECT_ID="$2"
+            # B-18 fix: PAT header.
             curl -sS -X POST https://backboard.railway.com/graphql/v2 \
                 -H "Content-Type: application/json" \
-                -H "Authorization: Bearer $TOKEN" \
+                -H "Project-Access-Token: $TOKEN" \
                 -d "{\"query\":\"query svcs(\$projectId: String!){ project(id:\$projectId){ services { edges { node { id name } } } } }\",\"variables\":{\"projectId\":\"$PROJECT_ID\"}}" \
                 | jq '[.data.project.services.edges[].node]'
           }
@@ -193,9 +200,10 @@ jobs:
                 --arg v "$VALUE" \
                 '{query:"mutation variableUpsert($input: VariableUpsertInput!) { variableUpsert(input: $input) }", variables:{input:{projectId:$p, environmentId:$e, serviceId:$s, name:$n, value:$v}}}')
             local RESP
+            # B-18 fix: PAT header.
             RESP=$(curl -sS -X POST https://backboard.railway.com/graphql/v2 \
                 -H "Content-Type: application/json" \
-                -H "Authorization: Bearer $TOKEN" \
+                -H "Project-Access-Token: $TOKEN" \
                 -d "$PAYLOAD")
             if echo "$RESP" | jq -e '.errors' > /dev/null; then
               echo "    upsert $NAME: ERR — $(echo "$RESP" | jq -c '.errors')"
@@ -216,9 +224,10 @@ jobs:
                 --arg e "$ENV_ID" \
                 '{query:"mutation serviceInstanceRedeploy($serviceId: String!, $environmentId: String!) { serviceInstanceRedeploy(serviceId: $serviceId, environmentId: $environmentId) }", variables:{serviceId:$s, environmentId:$e}}')
             local RESP
+            # B-18 fix: PAT header.
             RESP=$(curl -sS -X POST https://backboard.railway.com/graphql/v2 \
                 -H "Content-Type: application/json" \
-                -H "Authorization: Bearer $TOKEN" \
+                -H "Project-Access-Token: $TOKEN" \
                 -d "$PAYLOAD")
             if echo "$RESP" | jq -e '.errors' > /dev/null; then
               echo "    redeploy: ERR — $(echo "$RESP" | jq -c '.errors')"

